src/cephadm/cephadmlib/daemons/ceph.py

```diff
@@ -407,6 +410,55 @@ def create_daemon_dirs(self, data_dir: str, uid: int, gid: int) -> None:
         populate_files(ssl_dir, cert_files, uid, gid)
 
 
+def retrieve_version(container_id: str, container_command: str) -> str:
+    try:
+        exec_command = [
+            container_command,
+            'exec',
+            container_id,
+            'python3',
+            '-c',
+            "import sys; print(f'{sys.version_info.major}.{sys.version_info.minor}')",
+        ]
+
+        exec_result = subprocess.run(
+            exec_command, capture_output=True, text=True, check=True
+        )
```

Comment:
For consistency with other parts of cephadm I think you should use one of the functions in `call_wrappers` library.